packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/plugin.ts

```diff
@@ -56,23 +60,118 @@ export class WebpackAngularTypesPlugin {
-						this.addClassCodeDocDependency(ci, module);
+					for (const classInformation of classesInformation) {
+						this.addClassCodeDocDependency(classInformation, module);
 					}
-					for (const ii of interfacesInformation) {
-						this.addInterfaceCodeDocDependency(ii, module);
+					for (const interfaceInformation of interfacesInformation) {
+						this.addInterfaceCodeDocDependency(interfaceInformation, module);
+					}
+
+					const groupedExportsInformation = this.groupExportInformation(
+						functionsInformation,
+						constantsInformation,
+					);
+
+					for (const groupedExportInformation of groupedExportsInformation) {
+						this.addGroupedExportsCodeDocDependency(groupedExportInformation, module);
 					}
 				}
 				this.moduleQueue = [];
 			});
 		});
 	}
 
+	/**
+	 * Functions and constants can be annotated with "@group-docs" to collect them into a
+	 * single ArgsTable. Returns information grouped by the "groupBy" property.
+	 *
+	 * @param functionsInformation list of all functions
+	 * @param constantsInformation list of all constants
+	 */
+	private groupExportInformation(
+		functionsInformation: FunctionInformation[],
+		constantsInformation: ConstantInformation[],
+	): GroupedExportInformation[] {
+		const exportsInformationMap = new Map<string, ExportsInformation>();
+
+		for (const functionInformation of functionsInformation) {
+			const name = functionInformation.name;
+
+			exportsInformationMap.set(name, {
+				functionsInformation: [functionInformation],
+				constantsInformation: [],
+			});
+
+			const groupBys = functionInformation.groupBy;
+
+			for (const groupBy of groupBys) {
+				if (groupBy) {
+					const groupByEntry = exportsInformationMap.get(groupBy);
+					if (groupByEntry) {
+						if (groupByEntry.functionsInformation.length) {
+							groupByEntry.functionsInformation.push(functionInformation);
+						} else {
+							groupByEntry.functionsInformation = [functionInformation];
+						}
+					} else {
+						exportsInformationMap.set(groupBy, {
+							functionsInformation: [functionInformation],
+							constantsInformation: [],
+						});
+					}
+				}
+			}
+		}
+
+		for (const constantInformation of constantsInformation) {
+			const name = constantInformation.name;
+			const groupBys = constantInformation.groupBy;
+
+			exportsInformationMap.set(name, {
+				functionsInformation: [],
+				constantsInformation: [constantInformation],
+			});
+
+			for (const groupBy of groupBys) {
+				if (groupBy) {
+					const groupByEntry = exportsInformationMap.get(groupBy);
+					if (groupByEntry) {
+						if (groupByEntry.constantsInformation.length) {
+							groupByEntry.constantsInformation.push(constantInformation);
+						} else {
+							groupByEntry.constantsInformation = [constantInformation];
+						}
+					} else {
+						exportsInformationMap.set(groupBy, {
+							functionsInformation: [],
+							constantsInformation: [constantInformation],
+						});
+					}
+				}
+			}
+		}
```

Comment:
Same code as above. You could extract it into a method that takes an input of type `CommonConstantsLikeInformation` . Then you reuse the logic.